src/LocalRenderer/Window.cpp

```diff
@@ -0,0 +1,85 @@
+#include <GL/glew.h>
+#include <GLFW/glfw3.h>
+#include "Window.h"
+#include <stdexcept>
+#include <cstdio>
+
+const std::unordered_map<MouseButton, int> mouseButtonToGlfwButton = {
+    { MouseButton::Left,   GLFW_MOUSE_BUTTON_LEFT   },
+    { MouseButton::Right,  GLFW_MOUSE_BUTTON_RIGHT  },
+    { MouseButton::Middle, GLFW_MOUSE_BUTTON_MIDDLE },
+};
+
+Window::Window(std::uint32_t width, std::uint32_t height, const std::string& title)
+    : window_(nullptr, glfwDestroyWindow)
+    , width_(width)
+    , height_(height) {
+
+    if (!glfwInit()) {
+        throw std::runtime_error("glfwInit() failed");
+    }
+    // glfwWindowHint(GLFW_RESIZABLE, GLFW_FALSE);
+    window_.reset(glfwCreateWindow(width, height, title.c_str(), nullptr, nullptr));
+    if (!window_.get()) {
+        throw std::runtime_error("Failed to create GLFW window!");
+    }
+
+    glfwMakeContextCurrent(window_.get());
+    glfwSetWindowUserPointer(window_.get(), this);
+    glfwSetScrollCallback(window_.get(), scrollCallback);
+
+    GLenum glew_status = glewInit();
+
+    if (glew_status != GLEW_OK) {
```

Comment:
Just to save time with fixing the WSL issue that @bartektempka has mentioned, let's just check if a NO_GLX_DISPLAY error occurred.
```suggestion
    // GLEW_ERROR_NO_GLX_DISPLAY occurs when running the program on a WSL system.
    // This error is not critical and can be ignored.
    if (glew_status != GLEW_OK && glew_status != GLEW_ERROR_NO_GLX_DISPLAY) {
```